R/Reporter.R

```diff
@@ -302,16 +311,31 @@ Reporter <- R6::R6Class( # nolint: object_name_linter.
     #' reporter$from_jsondir(tmp_dir)
     from_jsondir = function(output_dir) {
       checkmate::assert_directory_exists(output_dir)
-      checkmate::assert_true(length(list.files(output_dir)) > 0)
       dir_files <- list.files(output_dir)
-      which_json <- grep("json$", dir_files)
-      json <- jsonlite::read_json(file.path(output_dir, dir_files[which_json]))
+      checkmate::assert_true(length(dir_files) > 0)
+      checkmate::assert_true("Report.json" %in% basename(dir_files))
```

Comment:
Please write more precise error messages
```suggestion
      checkmate::assert_true(length(dir_files) > 0)
      checkmate::assert_true("Report.json" %in% basename(dir_files))
```

Reply:
Improved